src/ui/src/builder/useComponentFieldViewModel.spec.ts

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { describe, it, expect, vi } from "vitest";
+import { ref, reactive, nextTick } from "vue";
+import { useComponentFieldViewModel } from "./useComponentFieldViewModel";
+
+type Component = { id: string; type: string; content: Record<string, string> };
+type Defs = Record<string, { fields?: Record<string, { default?: string }> }>;
+
+function mockDependencies(
+	initial: Record<string, Component>,
+	defs: Defs = {},
+): { components: Record<string, Component>; wf: any; ssbm: any } {
+	const components = reactive(initial);
+
+	function getComponentById(id: string) {
+		return components[id];
+	}
+
+	function getComponentDefinition(type?: string) {
+		return type ? defs[type] : undefined;
+	}
+
+	return {
+		components,
+		wf: {
+			getComponentById,
+			getComponentDefinition,
+			sendComponentUpdate: vi.fn(),
+		},
+		ssbm: {
+			openMutationTransaction: vi.fn(),
+			registerPreMutation: vi.fn(),
+			registerPostMutation: vi.fn(),
+			closeMutationTransaction: vi.fn(),
+		},
+	};
+}
+
+describe("useComponentFieldViewModel", () => {
+	const componentId = "TestComponent";
+	const fieldKey = "TestField";
+
+	function setupMockDependencies(componentId: string, fieldKey: string) {
+		return mockDependencies(
+			{ [componentId]: { id: componentId, type: "Text", content: {} } },
+			{ Text: { fields: { [fieldKey]: { default: "default test" } } } },
```

Comment:
You can use `buildMockComponent` which creates a valid button by default.

```suggestion
			{ [componentId]: buildMockComponent({ id: componentId, content: {} }) },
```